### PR TITLE
Add approve functions

### PR DESCRIPTION
**Pull request by dueyama**
I've implemented a test code to pages/index.tsx

An allowance value is max value of Uint256 now. I'm not sure it is OK or not.

components/metaMask/Approve.tsx is totally same to Omise-interface one.

We have to reset account on MetaMask every testing on Hardhat local node as following.
https://github.com/MetaMask/metamask-extension/issues/14963#issuecomment-1253676169

### CODE REVIEW

File: pages/index.tsx [hunk 1]
```diff
@@ -2,7 +2,8 @@ import type { NextPage } from 'next'
 import { Box, Container, Heading } from '@chakra-ui/react'
 import { useAccount, useConnect, useEnsName } from 'wagmi'
 import { InjectedConnector } from 'wagmi/connectors/injected'
-import { useMounted } from '@/hooks'
+import { Approve } from 'components/metaMask/Approve'
```
Comment (alecrem) [inline pages/index.tsx:5]: What do you think about renaming this to `components/Approve/index.tsx`?

I think it is more blockchain related than it is MetaMask related.

File: pages/index.tsx [hunk 2]
```diff
@@ -12,6 +13,14 @@ import useTranslation from 'next-translate/useTranslation'
 const Home: NextPage = () => {
   const isMounted = useMounted()
   const { t } = useTranslation('common')
+  const { address, isConnected } = useAccount()
+  const henkakuV2 = process.env
+    .NEXT_PUBLIC_CONTRACT_HENKAKUV2_ADDRESS as `0x${string}`
+  const nengajo = process.env
+    .NEXT_PUBLIC_CONTRACT_NENGAJO_ADDRESS as `0x${string}`
+  const { approved, allowanceValue } = useApproval(henkakuV2, nengajo, address)
```
Comment (alecrem) [inline pages/index.tsx:21]: `allowanceValue` is not used, so there is no need to caoture its value. I guess `{ approved }` should be enough.

The comment on the following line is not needed either.

**Review by alecrem — APPROVED** [review]
LGTM! This looks like it was a lot of work 👀

I made some comments recommending some simple changes, but if you are busy we can just merge and change that in a follow-up PR, or discuss my points later.

**Comment by dueyama** [comment]
thanks a lot! I will PR again in the morning.

File: pages/index.tsx (hunk 1 above)
Comment (dueyama) [reply pages/index.tsx:5]: I agree. Fixed.

File: pages/index.tsx (hunk 2 above)
Comment (dueyama) [reply pages/index.tsx:21]: Fixed.

**Comment by dueyama** [comment]
I fixed for all comments. Thanks!

**Comment by alecrem** [comment]
Thank you! Everything looks great. Let's merge this 😊

**Pull request merged**